`torch_wrapper` does not do inference properly on GPU

**Describe the bug**
When using an `ml_tools.torch_wrapper` instance to perform inference with a PyTorch model, the model itself is sent to GPU if `torch.cuda.is_available()`, but the input tensors are not, leading to a PyTorch error with tensors on different devices.

**To Reproduce**
1. Use analysis code such as the following (adapted from the processor example), paired with a simple MLP saved via TorchScript (that accepts 5 input features):
```
import logging
import numpy as np
import awkward as ak
from coffea import processor, ml_tools
from coffea.nanoevents import NanoEventsFactory, BaseSchema
from coffea.nanoevents.methods import candidate

class SimpleMLModule(ml_tools.torch_wrapper):
    def __init__(self, model_path):
        super().__init__(model_path)
    
    def prepare_awkward(self, events):
        muon = events.muon

        features = [
            muon.px,
            muon.py,
            muon.pz,
            muon.energy,
            muon.charge,
        ]

        inputs = {
            "x": ak.values_astype(ak.concatenate([x[:,np.newaxis] for x in features], axis=-1), "float32")
        }
        return (), inputs

class SimpleProcessor(processor.ProcessorABC):
    def __init__(self, mode="virtual"):
        assert mode in {"eager", "virtual", "dask"}
        self._mode = mode

        logging.debug("Instantiating Torch module.")
        self.model = SimpleMLModule("simple_mlp_torchscript.pt")
        logging.debug("Model instantiated.")
    
    def process(self, events):
        dataset = events.metadata["dataset"]
        muons = ak.zip(
            {
                "pt": events.Muon_pt,
                "eta": events.Muon_eta,
                "phi": events.Muon_phi,
                "mass": events.Muon_mass,
                "charge": events.Muon_charge,
            },
            with_name="PtEtaPhiMCandidate",
            behavior=candidate.behavior
        )

        cut = ak.num(muons) >= 1

        muon = muons[cut][:, 0]
        events = ak.with_field(events[cut], muon, "muon")

        logging.debug("Running ML model on events.")
        ml_fields = self.model(events)
        logging.debug(f"Model finished. Type of result: {ak.type(ml_fields)}")

        return {dataset: {"ml_stuff": ml_fields}}

    def postprocess(self, accumulator):
        pass

def main():
    filename = "Run2012B_DoubleMuParked.root"

    logging.debug("Loading events.")

    access_log = []
    events = NanoEventsFactory.from_root(
        {filename: "Events"},
        entry_stop = 10000,
        metadata = {"dataset": "DoubleMuon"},
        schemaclass = BaseSchema,
        mode = "virtual",
        access_log = access_log
    ).events()

    logging.debug("Creating Coffea processor")

    p = SimpleProcessor("virtual")

    logging.debug("Processing events.")
    out = p.process(events)
    logging.debug("Events processed.")
    
    # Just trying to print the result
    ml_fields = ak.to_numpy(out["DoubleMuon"]["ml_stuff"])
    logging.info(f"Got output from ML model. Shape: {ml_fields.shape}")

if (__name__ == "__main__"):
    logging.basicConfig(level=logging.DEBUG, format="[%(asctime)s - %(levelname)s]: %(message)s")
    logging.getLogger("fsspec.local").setLevel(logging.INFO) # suppress log every time a file is touched
    
    logging.info("Beginning Coffea processor test.")
    main()
    logging.info("Finished.")
```
2. Ensure that, in the Python environment, `torch.cuda.is_available()` evaluates to `True`.
3. Run the code, this produces a PyTorch error with the message: `RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!`

**Expected behavior**
I would expect the processor to run without error and inference to proceed on the GPU.

**Output**
N/A

**Desktop (please complete the following information):**
 - OS: Linux (AlmaLinux 8.10)
 - Coffea version: 2026.5
 - Torch version: 2.6.0_cu124
 - CUDA verison: 12.4

**Additional context**
I'm pretty sure this is a simple bug in the `numpy_call` method of `torch_wrapper`, please just let me know if I'm wrong. If i'm right, I would be happy to submit a patch for this :)

One can hit this with the pytest directly if one has the cuda supported torch installed (automatic on linux).
```
❯ pytest tests/test_ml_tools.py
=================================================================================================================================================== test session starts ====================================================================================================================================================
platform linux -- Python 3.14.4, pytest-9.0.3, pluggy-1.6.0 -- /home/iason/micromamba/envs/py3.14/bin/python3
cachedir: .pytest_cache
hypothesis profile 'default'
rootdir: /home/iason/Dropbox/work/pyhep_dev/coffea
configfile: pyproject.toml
plugins: xdist-3.8.0, cov-7.1.0, hypothesis-6.152.6, anyio-4.13.0, rerunfailures-16.3, timeout-2.4.0, doctestplus-1.7.1
collected 4 items

tests/test_ml_tools.py::test_triton SKIPPED (could not import 'tritonclient': No module named 'tritonclient')                                                                                                                                                                                                        [ 25%]
tests/test_ml_tools.py::test_torch FAILED                                                                                                                                                                                                                                                                            [ 50%]
tests/test_ml_tools.py::test_tensorflow SKIPPED (could not import 'tensorflow': No module named 'tensorflow')                                                                                                                                                                                                        [ 75%]
tests/test_ml_tools.py::test_xgboost PASSED                                                                                                                                                                                                                                                                          [100%]

========================================================================================================================================================= FAILURES =========================================================================================================================================================
________________________________________________________________________________________________________________________________________________________ test_torch ________________________________________________________________________________________________________________________________________________________

tmp_path = PosixPath('/tmp/pytest-of-iason/pytest-8/test_torch0')

    @pytest.mark.dask_client
    def test_torch(tmp_path):
        _ = pytest.importorskip("torch")

        from coffea.ml_tools.torch_wrapper import torch_wrapper

        client = Client()  # Spawn local cluster

        class torch_wrapper_test(torch_wrapper):
            def prepare_awkward(self, jets):
                default = common_prepare_awkward(jets)
                return [], {
                    "points": ak.values_astype(default["points"], np.float32),
                    "features": ak.values_astype(default["features"], np.float32),
                    "mask": ak.values_astype(default["mask"], np.float16),
                }

        tw = torch_wrapper_test("tests/samples/pn_demo.pt")
        ak_jets, dak_jets = prepare_jets_array(njets=256, tmp_path=tmp_path)
>       ak_res = tw(ak_jets)
                 ^^^^^^^^^^^

tests/test_ml_tools.py:157:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/coffea/ml_tools/helper.py:382: in __call__
    return self._call_awkward(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/coffea/ml_tools/helper.py:240: in _call_awkward
    np_rets = self._call_numpy(*np_args, **np_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/coffea/ml_tools/helper.py:181: in _call_numpy
    return self.numpy_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/coffea/ml_tools/torch_wrapper.py:130: in numpy_call
    return self.model(*args, **kwargs).detach().numpy()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/iason/micromamba/envs/py3.14/lib/python3.14/site-packages/torch/nn/modules/module.py:1778: in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = RecursiveScriptModule(
  original_name=ParticleNetTagger
  (conv): RecursiveScriptModule(
    original_name=FeatureCon...RecursiveScriptModule(original_name=Dropout)
      )
      (1): RecursiveScriptModule(original_name=Linear)
    )
  )
), args = ()
kwargs = {'features': tensor([[[ 0.5338,  0.3329,  0.7284,  ...,  0.0000,  0.0000,  0.0000],
         [-0.8538, -0.8412, -0.178...0,  0.3050,  ...,  0.0000,  0.0000,  0.0000],
         [ 0.4261,  0.0677, -0.0419,  ...,  0.0000,  0.0000,  0.0000]]])}

    def _call_impl(self, *args, **kwargs):
        forward_call = (self._slow_forward if torch._C._get_tracing_state() else self.forward)
        # If we don't have any hooks, we want to skip the rest of the logic in
        # this function, and just call forward.
        if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
                or _global_backward_pre_hooks or _global_backward_hooks
                or _global_forward_hooks or _global_forward_pre_hooks):
>           return forward_call(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           RuntimeError: The following operation failed in the TorchScript interpreter.
E           Traceback of TorchScript, serialized code (most recent call last):
E             File "code/__torch__/ParticleNet.py", line 14, in forward
E               mask: Tensor) -> Tensor:
E               conv = self.conv
E               features0 = (conv).forward(torch.mul(features, mask), )
E                            ~~~~~~~~~~~~~ <--- HERE
E               pn = self.pn
E               _0 = (pn).forward(points, features0, mask, )
E             File "code/__torch__/ParticleNet.py", line 27, in forward
E               x: Tensor) -> Tensor:
E               conv = self.conv
E               return (conv).forward(x, )
E                       ~~~~~~~~~~~~~ <--- HERE
E           class ParticleNet(Module):
E             __parameters__ = []
E             File "code/__torch__/torch/nn/modules/container.py", line 16, in forward
E               _2 = getattr(self, "2")
E               _3 = getattr(self, "3")
E               input0 = (_0).forward(input, )
E                         ~~~~~~~~~~~ <--- HERE
E               input1 = (_1).forward(input0, )
E               input2 = (_2).forward(input1, )
E             File "code/__torch__/torch/nn/modules/batchnorm.py", line 35, in forward
E               weight = self.weight
E               bias = self.bias
E               _3 = _0(input, running_mean, running_var, weight, bias, bn_training, 0.10000000000000001, 1.0000000000000001e-05, )
E                    ~~ <--- HERE
E               return _3
E             def _check_input_dim(self: __torch__.torch.nn.modules.batchnorm.BatchNorm1d,
E             File "code/__torch__/torch/nn/functional.py", line 14, in batch_norm
E             else:
E               pass
E             _2 = torch.batch_norm(input, weight, bias, running_mean, running_var, training, momentum, eps, True)
E                  ~~~~~~~~~~~~~~~~ <--- HERE
E             return _2
E           def relu(input: Tensor,
E
E           Traceback of TorchScript, original code (most recent call last):
E             File "/home/ensc/VirtualENV/ParticleNet.py", line 300, in forward
E
E                   points = points
E                   features = self.conv(features * mask)
E                              ~~~~~~~~~ <--- HERE
E                   mask = mask
E
E             File "/home/ensc/VirtualENV/ParticleNet.py", line 262, in forward
E               def forward(self, x):
E                   return self.conv(x)
E                          ~~~~~~~~~ <--- HERE
E             File "/home/ensc/VirtualENV/coffea-test/lib/python3.8/site-packages/torch/nn/modules/container.py", line 217, in forward
E               def forward(self, input):
E                   for module in self:
E                       input = module(input)
E                               ~~~~~~ <--- HERE
E                   return input
E             File "/home/ensc/VirtualENV/coffea-test/lib/python3.8/site-packages/torch/nn/modules/batchnorm.py", line 171, in forward
E                   used for normalization (i.e. in eval mode when buffers are not None).
E                   """
E                   return F.batch_norm(
E                          ~~~~~~~~~~~~ <--- HERE
E                       input,
E                       # If buffers are not to be tracked, ensure that they won't be updated
E             File "/home/ensc/VirtualENV/coffea-test/lib/python3.8/site-packages/torch/nn/functional.py", line 2450, in batch_norm
E                   _verify_batch_size(input.size())
E
E               return torch.batch_norm(
E                      ~~~~~~~~~~~~~~~~ <--- HERE
E                   input, weight, bias, running_mean, running_var, training, momentum, eps, torch.backends.cudnn.enabled
E               )
E           RuntimeError: Expected all tensors to be on the same device, but got weight is on cuda:0, different from other tensors on cpu (when checking argument in method wrapper_CUDA__native_batch_norm)

/home/iason/micromamba/envs/py3.14/lib/python3.14/site-packages/torch/nn/modules/module.py:1789: RuntimeError
---------------------------------------------------------------------------------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------------------------------------------------------------------------------
INFO     numexpr.utils:utils.py:164 NumExpr defaulting to 16 threads.
INFO     distributed.http.proxy:proxy.py:86 To route to workers diagnostics web server please install jupyter-server-proxy: python -m pip install jupyter-server-proxy
INFO     distributed.scheduler:scheduler.py:1754 State start
INFO     distributed.scheduler:scheduler.py:4209   Scheduler at:     tcp://127.0.0.1:34007
INFO     distributed.scheduler:scheduler.py:4224   dashboard at:  http://127.0.0.1:8787/status
INFO     distributed.scheduler:scheduler.py:8127 Registering Worker plugin shuffle
INFO     distributed.nanny:nanny.py:368         Start Nanny at: 'tcp://127.0.0.1:35999'
INFO     distributed.nanny:nanny.py:368         Start Nanny at: 'tcp://127.0.0.1:39239'
INFO     distributed.nanny:nanny.py:368         Start Nanny at: 'tcp://127.0.0.1:41173'
INFO     distributed.nanny:nanny.py:368         Start Nanny at: 'tcp://127.0.0.1:39997'
INFO     distributed.scheduler:scheduler.py:4563 Register worker addr: tcp://127.0.0.1:40819 name: 1
INFO     distributed.scheduler:scheduler.py:6181 Starting worker compute stream, tcp://127.0.0.1:40819
INFO     distributed.core:core.py:883 Starting established connection to tcp://127.0.0.1:40176
INFO     distributed.scheduler:scheduler.py:4563 Register worker addr: tcp://127.0.0.1:41869 name: 0
INFO     distributed.scheduler:scheduler.py:6181 Starting worker compute stream, tcp://127.0.0.1:41869
INFO     distributed.core:core.py:883 Starting established connection to tcp://127.0.0.1:40188
INFO     distributed.scheduler:scheduler.py:4563 Register worker addr: tcp://127.0.0.1:46425 name: 3
INFO     distributed.scheduler:scheduler.py:6181 Starting worker compute stream, tcp://127.0.0.1:46425
INFO     distributed.core:core.py:883 Starting established connection to tcp://127.0.0.1:40200
INFO     distributed.scheduler:scheduler.py:4563 Register worker addr: tcp://127.0.0.1:45419 name: 2
INFO     distributed.scheduler:scheduler.py:6181 Starting worker compute stream, tcp://127.0.0.1:45419
INFO     distributed.core:core.py:883 Starting established connection to tcp://127.0.0.1:40208
INFO     distributed.scheduler:scheduler.py:5916 Receive client connection: Client-29828cf9-637d-11f1-96b0-cc28aa466486
INFO     distributed.core:core.py:883 Starting established connection to tcp://127.0.0.1:40212
===================================================================================================================================================== warnings summary =====================================================================================================================================================
tests/test_ml_tools.py:85
  /home/iason/Dropbox/work/pyhep_dev/coffea/tests/test_ml_tools.py:85: PytestUnknownMarkWarning: Unknown pytest.mark.dask_client - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dask_client

tests/test_ml_tools.py:138
  /home/iason/Dropbox/work/pyhep_dev/coffea/tests/test_ml_tools.py:138: PytestUnknownMarkWarning: Unknown pytest.mark.dask_client - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dask_client

tests/test_ml_tools.py:184
  /home/iason/Dropbox/work/pyhep_dev/coffea/tests/test_ml_tools.py:184: PytestUnknownMarkWarning: Unknown pytest.mark.dask_client - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dask_client

tests/test_ml_tools.py:262
  /home/iason/Dropbox/work/pyhep_dev/coffea/tests/test_ml_tools.py:262: PytestUnknownMarkWarning: Unknown pytest.mark.dask_client - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dask_client

tests/test_ml_tools.py::test_torch
  /home/iason/Dropbox/work/pyhep_dev/coffea/src/coffea/ml_tools/helper.py:174: UserWarning: No format checks were performed on input!
    warnings.warn("No format checks were performed on input!")

tests/test_ml_tools.py::test_torch
  /home/iason/micromamba/envs/py3.14/lib/python3.14/site-packages/torch/jit/_serialization.py:170: DeprecationWarning: `torch.jit.load` is not supported in Python 3.14+ and may break. Please switch to `torch.export`.
    warnings.warn(

tests/test_ml_tools.py::test_xgboost
  /home/iason/micromamba/envs/py3.14/lib/python3.14/site-packages/distributed/node.py:187: UserWarning: Port 8787 is already in use.
  Perhaps you already have a cluster running?
  Hosting the HTTP server on port 46837 instead
    warnings.warn(

tests/test_ml_tools.py::test_xgboost
  /home/iason/micromamba/envs/py3.14/lib/python3.14/site-packages/xgboost/core.py:751: UserWarning: [16:01:06] WARNING: /__w/xgboost/xgboost/src/common/error_msg.cc:56: Empty dataset at worker: 0
    return func(**kwargs)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================================================================================= short test summary info ==================================================================================================================================================
FAILED tests/test_ml_tools.py::test_torch - RuntimeError: The following operation failed in the TorchScript interpreter.
SKIPPED [1] tests/test_ml_tools.py:87: could not import 'tritonclient': No module named 'tritonclient'
SKIPPED [1] tests/test_ml_tools.py:186: could not import 'tensorflow': No module named 'tensorflow'
==================================================================================================================================== 1 failed, 1 passed, 2 skipped, 8 warnings in 4.09s ====================================================================================================================================
```

--- a/src/coffea/ml_tools/torch_wrapper.py
+++ b/src/coffea/ml_tools/torch_wrapper.py
@@ -115,7 +115,7 @@ class torch_wrapper(nonserializable_attribute, numpy_call_wrapper):
                 torch.from_numpy(arr)
                 if arr.flags["WRITEABLE"]
                 else torch.from_numpy(numpy.copy(arr))
-            )
+            ).to(self.device)
             for arr in args
         ]
         kwargs = {
@@ -123,8 +123,8 @@ class torch_wrapper(nonserializable_attribute, numpy_call_wrapper):
                 torch.from_numpy(arr)
                 if arr.flags["WRITEABLE"]
                 else torch.from_numpy(numpy.copy(arr))
-            )
+            ).to(self.device)
             for key, arr in kwargs.items()
         }
         with torch.no_grad():
-            return self.model(*args, **kwargs).detach().numpy()
+            return self.model(*args, **kwargs).detach().cpu().numpy()

